etils/epath/gpath.py

```diff
@@ -270,6 +270,26 @@ def stat(self) -> stat_utils.StatResult:
     """Returns metadata for the file/directory."""
     return self._backend.stat(self._path_str)
 
+  def walk(
+      self: _P,
+      top_down: bool = True,
+      on_error: Optional[Callable] = None,
+      follow_symlinks: bool = False,
+      max_depth: Optional[int] = None,
+      **kwargs,
+  ) -> Iterator[_P, Iterable[_P], Iterable[_P]]:
+    for root, dirs, files in self._backend.walk(
+        top=self._path_str,
+        top_down=top_down,
+        on_error=on_error,
+        follow_symlinks=follow_symlinks,
+        max_depth=max_depth,
+        **kwargs,
+    ):
+      yield self._new(root), tuple(self._new(dir) for dir in dirs), tuple(
```

Comment:
Behavior should match `pathlib.Path.walk()` which returns `tuple[Self, list[str], list[str]`

Only `root` is self, `dir` and `files` are `list[str]`

Reply:
done